### PR TITLE
select preferred screen on startup; remember last dashboard screen used

### PR DESCRIPTION
**Pull request by brentpicasso**
select the preferred and last screen used.  "JUST WORKS"

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -395,15 +395,27 @@ def build(self):
...
             Clock.schedule_once(lambda dt: self.first_time_setup(), 0.5)
 
+    def show_startup_view(self):
+        settings_to_view = {'Home Page':'home',
+                            'Dashboard':'dash',
+                            'Analysis': 'analysis',
+                            'Configuration': 'config' }
+        try:
+            view_pref = self.settings.userPrefs.get_pref('preferences', 'startup_screen')
+            view = settings_to_view[view_pref]
```
Comment (ryandoherty) [inline main.py]: We should handle the case here of `get_pref` returning None or False if the setting hasn't been set yet or the prefs file is gone.

File: main.py [hunk 2]
```diff
@@ -395,15 +395,27 @@ def build(self):
...
+            self.showMainView(view)
+        except:
+            Logger.error("RaceCaptureApp: could not show main view " + str(view_pref))
```
Comment (ryandoherty) [inline main.py]: For exceptions, should we log the stack trace or more info as to why the exception occurred? Also I'm wondering if we should start pushing any caught exceptions to Sentry so we can see if there's pieces of the app that aren't working for users.

Comment (brentpicasso) [inline main.py]: I kept it brief b/c the likely case is bad dictionary key. 

Yes, a central handler might be good for non-benign errors!

File: main.py (hunk 1 above)
Comment (brentpicasso) [inline main.py]: I think we're ok b/c  preferences automatically handles default values.

File: autosportlabs/racecapture/views/dashboard/dashboardview.py [hunk 3]
```diff
@@ -79,12 +81,23 @@ def init_view(self):
...
+
+    def _show_screen(self, screen):
+        self._screen_mgr.current = screen
+        self._settings.userPrefs.set_pref('preferences', 'last_dash_screen', screen)
+
+    def _show_last_view(self):
+        try:
+            last_screen_name = self._settings.userPrefs.get_pref('preferences', 'last_dash_screen')
+            self._screen_mgr.current = last_screen_name
+        except:
```
Comment (ryandoherty) [inline autosportlabs/racecapture/views/dashboard/dashboardview.py]: Should we do the same thing we did in main.py and remove the try/except and let the userPrefs return the default if the preference isn't found?

Comment (brentpicasso) [inline autosportlabs/racecapture/views/dashboard/dashboardview.py]: yup, tested and letting the crash handler pick it up works fine.

**Comment by ryandoherty** [comment]
:shipit:

**Pull request merged**